### PR TITLE
[Ready]Adds 2 new quirks! Quick step and Light Sensitivity 

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
[Changelogs]
2 New Quirks!
Light Sensitivity - Well in light you feel bad give 1 point
Quick step - You walk faster cost 2 points

[why]
Was asked to make these works and Cydia helped make them!
K5 and Cydia helped me make them! What good friends,

### CODE REVIEW

**Comment by king5327** [comment]
In all fairness, the speed boost could be a bit smaller.

File: code/modules/mob/living/carbon/human/species.dm [hunk 1]
```diff
@@ -1221,6 +1221,9 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 		. += speedmod
 		. += H.physiology.speed_mod
 
+	if(H.has_trait(TRAIT_SPEEDY_STEP))
+		. -= 0.5
```
Comment (deathride58) [inline code/modules/mob/living/carbon/human/species.dm]: No. Especially not with the current tile-based & delay-based movement system.

**Review by deathride58 — DISMISSED** [review]

**Comment by deathride58** [comment]
A better alternative to a flat movement speed increase across the board would be to just increase walking speed a smidge when the trait's present

**Comment by JogBoba** [comment]
I like the idea of these in theory, though with the effects mood has on gameplay and the fact that sanity comes into play, I would think that light sensitivity(depending on how bright the light needs to be) would be a bit too large of an impediment for just one point.

File: code/modules/mob/living/carbon/human/species.dm [hunk 2]
```diff
@@ -1221,6 +1221,9 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 		. += speedmod
 		. += H.physiology.speed_mod
 
+	if (H.m_intent == MOVEMENT_INTENT_WALK && has_trait("quickie_feet"))
```
Comment (Ghommie) [inline code/modules/mob/living/carbon/human/species.dm]:
```suggestion
	if (H.m_intent == MOVEMENT_INTENT_WALK && H.has_trait(TRAIT_SPEEDY_STEP))
```

Also check if this isn't overridden by the subtypes.

File: code/modules/mob/living/carbon/human/species.dm [hunk 3]
```diff
@@ -1221,6 +1221,9 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 		. += speedmod
 		. += H.physiology.speed_mod
 
+	if (H.m_intent == MOVEMENT_INTENT_WALK && H.has_trait(TRAIT_SPEEDY_STEP))
```
Comment (deathride58) [inline code/modules/mob/living/carbon/human/species.dm]:
```suggestion
	if (H.m_intent == MOVE_INTENT_WALK && H.has_trait(TRAIT_SPEEDY_STEP))
```

**Comment by MimicFaux** [comment]
To be clear, is this increasing walk speed, or run speed slightly?
One would be pleasant tweak, the other would be almost universally taken.

**Comment by deathride58** [comment]
It's a walk speed increase, run speed is unaffected

File: code/datums/traits/good.dm [hunk 4]
```diff
@@ -58,6 +58,14 @@
 	gain_text = "<span class='notice'>You walk with a little more litheness.</span>"
 	lose_text = "<span class='danger'>You start tromping around like a barbarian.</span>"
 
+/datum/quirk/quick_step
+	name = "Quick Step"
+	desc = "You walk with a bounce in you'r step; Making you slightly faster then most other crewmembers."
```
Comment (Ragolution) [inline code/datums/traits/good.dm]:
```suggestion
	desc = "You walk with determined strides, and out-pace most people when walking."
```

File: code/datums/traits/good.dm [hunk 5]
```diff
@@ -58,6 +58,14 @@
...
+	value = 2
+	mob_trait = TRAIT_SPEEDY_STEP
+	gain_text = "<span class='notice'>You feel quick on your feet.</span>"
```
Comment (Ragolution) [inline code/datums/traits/good.dm]:
```suggestion
	gain_text = "<span class='notice'>You feel determined. No time to lose.</span>"
```

File: code/datums/traits/good.dm [hunk 6]
```diff
@@ -58,6 +58,14 @@
...
+	lose_text = "<span class='danger'>You feel a little more slow.</span>"
```
Comment (Ragolution) [inline code/datums/traits/good.dm]:
```suggestion
	lose_text = "<span class='danger'>You feel less determined. What's the rush, man?</span>"
```

File: code/datums/traits/negative.dm [hunk 7]
```diff
@@ -144,6 +144,19 @@
 	else
 		SEND_SIGNAL(quirk_holder, COMSIG_CLEAR_MOOD_EVENT, "nyctophobia")
 
+/datum/quirk/lightless
+	name = "Light Sensitivity"
+	desc = "Any type of light just makes you feel uneasy, it's hard to keep in a good mood when your sink itches and eyes burn from light!"
```
Comment (Ragolution) [inline code/datums/traits/negative.dm]: I too hate when my sink itches.

Comment (Ragolution) [inline code/datums/traits/negative.dm]:
```suggestion
	desc = "Bright lights irritate you. Your eyes start to water, your skin feels itchy against the photon radiation, and your hair gets dry and frizzy. Maybe it's a medical condition. If only Nanotrasen was more considerate of your needs..."
```

File: code/datums/traits/negative.dm [hunk 8]
```diff
@@ -144,6 +144,19 @@
 	else
 		SEND_SIGNAL(quirk_holder, COMSIG_CLEAR_MOOD_EVENT, "nyctophobia")
 
+/datum/quirk/lightless
```
Comment (Ragolution) [inline code/datums/traits/negative.dm:147]: Needs gain/loss messages

**Review by Ragolution — CHANGES_REQUESTED** [review]

**Review by silicons — CHANGES_REQUESTED** [review]
Put the check in base mob movement and use the movement modification system if it exists here.

**Review by silicons — APPROVED** [review]

**Pull request merged**